### DISCUSSION TITLE
How to get previous value in signals?

### DISCUSSION THREAD

**Discussion by ganjarsetia** [preactjs/signals#309] [Q&A] [answered]
When I'm using `useState` hooks I can get previous state, like this code below

```jsx
const [object, setObject] = useState({
  firstKey: '',
  secondKey: '',
});

setObject((prevState) => ({
  ...prevState,
  secondKey: 'value',
}));
```

Ho to do like this using signals?

**Comment by rschristian** [comment, marked as answer]
It's just normal JS, no need to do anything special:

```js
const myObject = signal({
    firstKey: '',
    secondKey: '',
});

myObject.value = { ...myObject.value, secondKey: 'value' };
```

**Reply by ganjarsetia** [reply]
Thank you!

I was thinking Preact signals has its own way of getting previous value like in SolidJS Signal:

```jsx
const [route, setRoute] = createSignal(1);

createEffect((prev) => {
  const updated = route();

  console.log('route previous value is', prev);
  console.log('route updated value is', updated);

  return updated;
}, route());
```

turn out It's just normal JS

**Comment by hopemanryan** [comment]
> When I'm using `useState` hooks I can get previous state, like this code below
> 
> 
> 
> ```jsx
> 
> const [object, setObject] = useState({
> 
>   firstKey: '',
> 
>   secondKey: '',
> 
> });
> 
> 
> 
> setObject((prevState) => ({
> 
>   ...prevState,
> 
>   secondKey: 'value',
> 
> }));
> 
> ```
> 
> 
> 
> Ho to do like this using signals?

So I actually started a fork with pr to add a history array for previous values. 
It quickly became a need for me
